src/map/src/combat/player.cpp

```diff
@@ -74,3 +76,12 @@ void Player::toggle_player_move(EntitySystem& entitySystem, RoseCommon::Entity e
 	pToggle.set_object_id(basicInfo.id);
 	entitySystem.send_map(pToggle);
 }
+
+void Player::set_animation(EntitySystem& entitySystem, RoseCommon::Entity entity, const RoseCommon::Packet::CliSetAnimation& packet) {
+	auto& basicInfo = entitySystem.get_component<Component::BasicInfo>(entity);
+	auto pAnimation = Packet::SrvSetAnimation::create();
+	pAnimation.set_id(packet.get_id());
+	pAnimation.set_value(packet.get_value());
+	pAnimation.set_object_id(basicInfo.id);
+	entitySystem.send_nearby(entity, pAnimation);
```

Comment:
We will probably want to add some checks here at some point, to make sure the clients have the animation / are not trying to hack

Reply:
Client is handling all those checks. whether there is no motion it just doing nothing :)